torii/src/lib.rs

```diff
@@ -85,177 +94,182 @@ impl Torii {
-            .and_then(|| async { Ok::<_, Infallible>(routing::handle_health()) });
-
-        let get_router = warp::get().and(
-            warp::path(uri::CONFIGURATION)
-                .and(add_state!(self.kiso))
-                .and_then(|kiso| async move {
-                    Ok::<_, Infallible>(WarpResult(routing::handle_get_configuration(kiso).await))
+    fn create_api_router(&self) -> axum::Router {
+        let router = Router::new()
+            .route(uri::HEALTH, get(routing::handle_health))
+            .route(
+                uri::CONFIGURATION,
+                get({
+                    let kiso = self.kiso.clone();
+                    move || routing::handle_get_configuration(kiso)
                 }),
-        );
+            );
 
         #[cfg(feature = "telemetry")]
-        let get_router = get_router
-            .or(warp::path(uri::STATUS)
-                .and(add_state!(self.metrics_reporter.clone()))
-                .and(warp::header::optional(warp::http::header::ACCEPT.as_str()))
-                .and(warp::path::tail())
-                .and_then(
-                    |metrics_reporter, accept: Option<String>, tail| async move {
-                        Ok::<_, Infallible>(crate::utils::WarpResult(routing::handle_status(
+        let router = router
+            .route(
+                &format!("{}/*tail", uri::STATUS),
+                get({
+                    let metrics_reporter = self.metrics_reporter.clone();
+                    move |headers: axum::http::header::HeaderMap, axum::extract::Path(tail): axum::extract::Path<String>| {
+                        let accept = headers.get(axum::http::header::ACCEPT);
+                        core::future::ready(routing::handle_status(
                             &metrics_reporter,
-                            accept.as_ref(),
-                            &tail,
-                        )))
-                    },
-                ))
-            .or(warp::path(uri::METRICS)
-                .and(add_state!(self.metrics_reporter))
-                .and_then(|metrics_reporter| async move {
-                    Ok::<_, Infallible>(crate::utils::WarpResult(routing::handle_metrics(
-                        &metrics_reporter,
-                    )))
-                }))
-            .or(warp::path(uri::API_VERSION)
-                .and(add_state!(self.state.clone()))
-                .and_then(|state| async {
-                    Ok::<_, Infallible>(routing::handle_version(state).await)
-                }));
+                            accept,
+                            Some(&tail),
+                        ))
+                    }
+                }),
+            )
+            .route(
+                uri::STATUS,
+                get({
+                    let metrics_reporter = self.metrics_reporter.clone();
+                    move |headers: axum::http::header::HeaderMap| {
+                        let accept = headers.get(axum::http::header::ACCEPT);
```

Comment:
I see this is repeated several times.. Maybe it's worth making an `Accept` [extractor](https://docs.rs/axum/latest/axum/extract/index.html#defining-custom-extractors)?

UPD: or we can go one step further and make something akin to https://docs.rs/axum-content-negotiation/ for json/scale responses as I remember being discussed some time ago, though that is probably for a separate PR

Reply:
i've added `ExtractAccept` header, would like to handle content negotiation in separate PR